test/addons/at-exit/binding.cc

```diff
@@ -10,33 +10,25 @@ using v8::Local;
 using v8::Object;
 
 static char cookie[] = "yum yum";
-static int at_exit_cb1_called = 0;
-static int at_exit_cb2_called = 0;
 
 static void at_exit_cb1(void* arg) {
   Isolate* isolate = static_cast<Isolate*>(arg);
   HandleScope handle_scope(isolate);
   Local<Object> obj = Object::New(isolate);
   assert(!obj.IsEmpty());  // Assert VM is still alive.
   assert(obj->IsObject());
-  at_exit_cb1_called++;
+  fprintf(stderr, "at_exit_cb1\n");
 }
 
 static void at_exit_cb2(void* arg) {
   assert(arg == static_cast<void*>(cookie));
-  at_exit_cb2_called++;
-}
-
-static void sanity_check(void) {
-  assert(at_exit_cb1_called == 1);
-  assert(at_exit_cb2_called == 2);
+  fprintf(stderr, "at_exit_cb2\n");
 }
 
 void init(Local<Object> exports) {
   AtExit(at_exit_cb1, exports->GetIsolate());
   AtExit(at_exit_cb2, cookie);
   AtExit(at_exit_cb2, cookie);
-  atexit(sanity_check);
```

Comment:
Would a static destructor also solve the problem? That would run at unload time

Reply:
I think this is the best alternative.